### PR TITLE
 Implementing an abstract NetworkManager

### PR DESCRIPTION
**Pull request by AnnaBanani**
class NetworkManager created
func loadData created

### CODE REVIEW

File: TheMET/TheMET/Network/NetworkManager.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class NetworkManager {
+    
+    func loadData(urlString: String, parameters: [String : String], completion: @escaping([String: Any]?) -> Void)  {
+        var components = URLComponents(string: urlString)!
+        components.queryItems = parameters.map{ (key, value) in
+            URLQueryItem(name: key, value: value)
+        }
+        components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
+        let urlRequest = URLRequest(url: components.url!)
```
Comment (Ssimboss) [inline TheMET/TheMET/Network/NetworkManager.swift:18]: Do not force unwrap `components.url!`. Instead, unwrap it properly and execute `completion` like you do for an error.

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/Network/NetworkManager.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//
+//  Created by Анна Ситникова on 21/03/2023.
+//
+
+import Foundation
+
+class NetworkManager {
+    
+    func loadData(urlString: String, parameters: [String : String], completion: @escaping([String: Any]?) -> Void)  {
+        var components = URLComponents(string: urlString)!
```
[the remark this answers is not in the document]
Comment (AnnaBanani) [reply TheMET/TheMET/Network/NetworkManager.swift:13]: Fixed

File: TheMET/TheMET/Network/NetworkManager.swift (hunk 1 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Network/NetworkManager.swift:18]: Fixed

**Review by Ssimboss — CHANGES_REQUESTED** [review]

**Review by Ssimboss — APPROVED** [review]
Please resolve the conflict and merge the branch

**Pull request merged**